src/key_value_store/core/KeyValueStore.cpp

```diff
@@ -0,0 +1,66 @@
+#include "KeyValueStore.h"
+
+#include <mutex>
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+// Alias for the unorderedList
+typedef std::unordered_map<std::string, 
+  std::vector<std::string>> hashMap; 
+
+bool KeyValueStore::Put(std::string key, std::string value) {
+  bool result; 
+  lock_.lock();
+  // If this key already exists in the store
+  if (kv_store_.find(key) != kv_store_.end()) {
+    // If user puts the same function
+    if (CheckIfExists(kv_store_[key], value)) {
+      result = false;
+    } else {
+      int size = kv_store_[key].size();
+      kv_store_[key].push_back(value);
+      // If the value was not inserted
+      if (size >= kv_store_[key].size()) {
+        result = false;
+      } else {
+        result = true;
+      }
+    }
+  } else { // If they key doesn't exist
+    //Create a new vector with `value`
+    result = kv_store_.insert({key, {value}}).second;
```

Comment:
You don't have to explicitly create a new vector -- if you try to insert into it push_back() with the bracket operator on the map, it'll default construct it.

Reply:
Yup, I know, but this way insert returns a pair <iterator, boolean> that tells me if the operation was successful or not. So doing .second will equal a boolean value representing success/failure.